### PR TITLE
feat(provider/docker): Added insecure registry support

### PR DESCRIPTION
**Pull request by podnov**
Added an insecureRegistry flag to clouddriver-docker registry
support that will skip SSL certificate validation.

This will allow for addressing spinnaker/halyard#680.

### CODE REVIEW

**Comment by podnov** [comment]
I'd love to be able to write a better test for this case, but can't think of a good way short of having a publicly accessible insecure registry.

**Comment by edwinavalos** [comment]
@lwander wander says, "Tell him @lwanders is very happy to see this"

We did run into a similar issue at Target, in the end we decided to create custom containers and specified them via artifactId, https://www.spinnaker.io/reference/halyard/custom/#tweakable-service-settings

File: clouddriver-docker/src/main/groovy/com/netflix/spinnaker/clouddriver/docker/registry/api/v2/client/DockerRegistryClient.groovy [hunk 1]
```diff
@@ -92,16 +97,21 @@ class DockerRegistryClient {
       return this
     }
 
+    Builder insecureRegistry(boolean insecureRegistry) {
```
Comment (lwander) [inline clouddriver-docker/src/main/groovy/com/netflix/spinnaker/clouddriver/docker/registry/api/v2/client/DockerRegistryClient.groovy]: I see why the halyard build failed - we need to release this change to clouddriver once it's merged - I'll push it through

**Comment by lwander** [comment]
BTW thanks for this :)

**Comment by podnov** [comment]
My pleasure :)

File: clouddriver-docker/src/main/groovy/com/netflix/spinnaker/clouddriver/docker/registry/api/v2/client/DockerRegistryClient.groovy [hunk 2]
```diff
@@ -149,14 +167,14 @@ class DockerRegistryClient {
     this.catalogFile = catalogFile
   }
 
-  DockerRegistryClient(String address, String email, String username, String password, long clientTimeoutMillis, int paginateSize, String catalogFile) {
-    this(address, clientTimeoutMillis, paginateSize)
+  DockerRegistryClient(String address, String email, String username, String password, long clientTimeoutMillis, int paginateSize, String catalogFile, boolean insecureRegistry) {
+    this(address, clientTimeoutMillis, paginateSize, catalogFile, insecureRegistry)
```
Comment (podnov) [inline clouddriver-docker/src/main/groovy/com/netflix/spinnaker/clouddriver/docker/registry/api/v2/client/DockerRegistryClient.groovy]: I think f70d6ea introduces a potential bug as it's not passing catalogFile to the underlying constructor.

File: clouddriver-docker/src/main/groovy/com/netflix/spinnaker/clouddriver/docker/registry/api/v2/client/DockerRegistryClient.groovy [hunk 3]
```diff
@@ -149,14 +167,14 @@ class DockerRegistryClient {
...
     this.tokenService = new DockerBearerTokenService(username, password)
     this.email = email
   }
 
-  DockerRegistryClient(String address, String email, String username, File passwordFile, long clientTimeoutMillis, int paginateSize, String catalogFile) {
-    this(address, clientTimeoutMillis, paginateSize)
+  DockerRegistryClient(String address, String email, String username, File passwordFile, long clientTimeoutMillis, int paginateSize, String catalogFile, boolean insecureRegistry) {
+    this(address, clientTimeoutMillis, paginateSize, catalogFile, insecureRegistry)
```
Comment (podnov) [inline clouddriver-docker/src/main/groovy/com/netflix/spinnaker/clouddriver/docker/registry/api/v2/client/DockerRegistryClient.groovy]: I think f70d6ea introduces a potential bug as it's not passing catalogFile to the underlying constructor.

File: clouddriver-docker/src/main/groovy/com/netflix/spinnaker/clouddriver/docker/registry/api/v2/client/DockerRegistryClient.groovy (hunk 2 above)
Comment (ethanfrogers) [reply clouddriver-docker/src/main/groovy/com/netflix/spinnaker/clouddriver/docker/registry/api/v2/client/DockerRegistryClient.groovy]: @podnov I just merged this last night. I'll double check that. If it's missing, it was an oversight! Thanks for the heads up!

Comment (ethanfrogers) [reply clouddriver-docker/src/main/groovy/com/netflix/spinnaker/clouddriver/docker/registry/api/v2/client/DockerRegistryClient.groovy]: @podnov fixed in #1903

**Comment by podnov** [comment]
Anything I can do to help getting this merged?

**Comment by lwander** [comment]
Sorry for the wait, I took another read & looked at the trust manager docs. LGTM

**Comment by lwander** [comment]
I will cut a release of clouddriver, update spinnaker-dependencies & then show you how to integrate this change into clouddriver.

**Comment by lwander** [comment]
Actually... there's a bug in clouddriver k8s initialization that I don't want to pull into halyard - let's get https://github.com/spinnaker/clouddriver/pull/1933 resolved first.

**Pull request merged**